vue-app/src/components/Transaction.vue

```diff
@@ -2,13 +2,22 @@
   <div class="transaction">
     <template v-if="error">
       <div class="error">{{ error }}</div>
-      <button
-        v-if="displayCloseBtn"
-        class="btn-secondary close-btn"
-        @click="$emit('close')"
-      >
-        Close
-      </button>
+      <div class="btn-row">
+        <button
+          v-if="displayCloseBtn"
+          class="btn-secondary close-btn"
+          @click="$emit('close')"
+        >
+          Close
+        </button>
+        <button
+          v-if="displayRetryBtn"
+          class="btn-primary close-btn"
+          @click="retryMethod()"
```

Comment:
You can use the builtin event props in this case and we could have a cleaner interface in this component (as we are doing with the close event in the close button).
We could do `@click="$emit('retry')"` here, and from the consumer:
```
<transaction @retry="doSomething">
```

More about this here: https://vuejs.org/v2/guide/components-custom-events.html

Reply:
Cool!